src/app/leverage/provider.tsx

```diff
@@ -239,6 +245,47 @@ export function LeverageProvider(props: { children: any }) {
     fetchStats()
   }, [baseToken])
 
+  useEffect(() => {
+    async function fetchCostOfCarry(
+      jsonRpcProvider: providers.JsonRpcProvider,
```

Comment:
Not sure if there's a way around using ethers as it's called out in their docs

Reply:
If the SDK doesn't provide adapters for viem, it's probably built on ethers.